xarray/namedarray/core.py

```diff
@@ -834,6 +834,7 @@ def chunk(
         if chunkmanager.is_chunked_array(data_old):
             data_chunked = chunkmanager.rechunk(data_old, chunks)  # type: ignore[arg-type]
         else:
+            ndata: duckarray[Any, Any]
```

Comment:
I removed the pandas-specific code, I'm not sure we should do that, we might as well just ask the user to cast.